compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -959,25 +965,49 @@ execMergeDars darFp1 darFp2 mbOutFp =
         pure $ ZipArchive.toEntry manifestPath 0 $ BSLC.unlines $
             map (\(k, v) -> breakAt72Bytes $ BSL.fromStrict $ k <> ": " <> v) attrs1
 
-execDocTest :: Options -> [FilePath] -> Command
-execDocTest opts files =
+execDocTest :: Options -> FilePath -> Bool -> [FilePath] -> Command
+execDocTest opts scriptDar noSource files =
   Command DocTest Nothing effect
   where
     effect = do
       let files' = map toNormalizedFilePath' files
+          packageFlag =
+            ExposePackage
+              ("--package daml-script-" <> SdkVersion.sdkPackageVersion)
+              (UnitIdArg $ stringToUnitId $ "daml-script-" <> SdkVersion.sdkPackageVersion)
+              (ModRenaming True [])
+
       logger <- getLogger opts "doctest"
+
+      -- Install daml-script in their project and update the package db
+      -- TODO: Should this be "."?
+      installDependencies "." opts (PackageSdkVersion SdkVersion.sdkVersion) [scriptDar] []
+      createProjectPackageDb "." opts mempty
+
+      opts <- pure opts
+        { optPackageDbs = projectPackageDatabase : optPackageDbs opts
+        , optPackageImports = packageFlag : optPackageImports opts
+        -- Drop version information to avoid package clashes (specifically when generating for internal packages)
```

Comment:
Consider maybe only doing this for `no-source`? Its somewhat misleading though, and doesn't break non-stdlib cases